src/main/java/mate/academy/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package mate.academy.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.user.UserRegistrationRequestDto;
+import mate.academy.dto.user.UserResponseDto;
+import mate.academy.exception.RegistrationException;
+import mate.academy.mapper.UserMapper;
+import mate.academy.model.User;
+import mate.academy.repository.UserRepository;
+import mate.academy.service.UserService;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```

Comment:
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```
It's better to use this method to avoid calling isPresent()